### PR TITLE
add ESModule template

### PR DESCRIPTION
**Pull request by mtosi**
as requested by the TSG,
https://docs.google.com/document/d/1ilqSm8_6ImzK3RR3bxLemjri-4yXQAt9bWKvo-BYd2U/edit#heading=h.948g1p3wtuig

among different ESModules used @HLT,
I found there were few cfi files missing

please, find below the list of modules I checked @gennai @fwyzard 

PropagatorWithMaterialESProducer           --> TrackingTools/MaterialEffects/python/MaterialPropagator_cfi.py
ParametrizedMagneticFieldProducer          --> MagneticField/ParametrizedEngine/python/ParabolicParametrizedField_cfi.py
Chi2MeasurementEstimatorESProducer         --> TrackingTools/KalmanUpdators/python/Chi2MeasurementEstimatorESProducer_cfi.py
Chi2ChargeMeasurementEstimatorESProducer   --> RecoTracker/MeasurementDet/python/Chi2ChargeMeasurementEstimatorESProducer_cfi.py
AnalyticalPropagatorESProducer             --> TrackingTools/Producers/python/analyticalPropagatorESProducer_cfi.py NEW
ClusterShapeHitFilterESProducer            --> RecoPixelVertexing/PixelLowPtUtilities/python/ClusterShapeHitFilterESProducer_cfi.py
SiStripGainESProducer                      --> CalibTracker/SiStripESProducers/python/SiStripGainESProducer_cfi.py
SiStripQualityESProducer                   --> CalibTracker/SiStripESProducers/python/SiStripQualityESProducer_cfi.py
SiStripRecHitMatcherESProducer             --> RecoLocalTracker/SiStripRecHitConverter/python/SiStripRecHitMatcher_cfi.py
SteppingHelixPropagatorESProducer          --> TrackPropagation/SteppingHelixPropagator/python/SteppingHelixPropagator_cfi.py
TrackerDigiGeometryESModule                --> Geometry/TrackerGeometryBuilder/python/trackerDigiGeometryESModule_cfi.py
TrackerGeometricDetESModule                --> Geometry/TrackerNumberingBuilder/test/tracker_cfg.py
TransientTrackBuilderESProducer            --> TrackingTools/TransientTrack/python/TransientTrackBuilder_cfi.py
NavigationSchoolESProducer                 --> RecoTracker/TkNavigation/python/NavigationSchoolESProducer_cfi.py
KFFittingSmootherESProducer                --> TrackingTools/TrackFitters/python/KFFittingSmootherESProducer_cfi.py
GlobalTrackingGeometryESProducer           --> Geometry/GlobalTrackingGeometryBuilder/python/globalTrackingGeometryESProducer_cfi.py NEW
KFTrajectoryFitterESProducer               --> TrackingTools/TrackFitters/python/KFTrajectoryFitterESProducer_cfi.py
KFUpdatorESProducer                        --> TrackingTools/KalmanUpdators/python/KFUpdatorESProducer_cfi.py
MeasurementTrackerESProducer               --> RecoTracker/MeasurementDet/python/MeasurementTrackerESProducer_cfi.py
PixelCPEGenericESProducer                  --> RecoLocalTracker/SiPixelRecHits/python/PixelCPEGeneric_cfi.py
StripCPEESProducer                         --> RecoLocalTracker/SiStripRecHitConverter/python/StripCPEESProducer_cfi.py
TkTransientTrackingRecHitBuilderESProducer --> RecoTracker/TransientTrackingRecHit/python/TransientTrackingRecHitBuilder_cfi.py
TrackerRecoGeometryESProducer              --> RecoTracker/GeometryESProducer/python/TrackerRecoGeometryESProducer_cfi.py
TrajectoryCleanerESProducer                --> TrackingTools/Producers/python/trajectoryCleanerESProducer_cfi.py NEW
SiPixelQualityESProducer                   --> CalibTracker/SiPixelESProducers/python/SiPixelQualityESProducer_cfi.py
SiPixelTemplateDBObjectESProducer          --> CalibTracker/SiPixelESProducers/python/SiPixelTemplateDBObjectESProducer_cfi.py
SiStripBackPlaneCorrectionDepESProducer    --> CalibTracker/SiStripESProducers/python/SiStripBackPlaneCorrectionDepESProducer_cfi.py
SiStripLorentzAngleDepESProducer           --> CalibTracker/SiStripESProducers/python/SiStripLorentzAngleDepESProducer_cfi.py
SiStripConnectivity                        --> CalibTracker/SiStripESProducers/python/SiStripConnectivity_cfi.py

### CODE REVIEW

**Comment by mtosi** [comment]
I'm sorry, but I really do not understand how such a PR can produce this errors !
what is going on w/ PR testing in the last days ?

**Comment by ktf** [comment]
Yeah, It's unrelated. @nclopezo can you please start again all the tests 
which were failing because of this?

File: Geometry/GlobalTrackingGeometryBuilder/python/globalTrackingGeometryESProducer_cfi.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import FWCore.ParameterSet.Config as cms
+
+globalTrackingGeometryESProducer = cms.ESProducer( "GlobalTrackingGeometryESProducer" )
```
Comment (ianna) [inline Geometry/GlobalTrackingGeometryBuilder/python/globalTrackingGeometryESProducer_cfi.py]: This is not missing file - the bare producer is defined here:
http://cmslxr.fnal.gov/source/Geometry/CommonDetUnit/python/bareGlobalTrackingGeometry_cfi.py?v=CMSSW_7_4_X_2015-01-15-0200#0010

File: Geometry/TrackerGeometryBuilder/python/trackerDigiGeometryESModule_cfi.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    ROCS_Y = cms.int32( 0 ),
+    upgradeGeometry = cms.bool( False ),
+    BIG_PIX_PER_ROC_Y = cms.int32( 2 ),
+    BIG_PIX_PER_ROC_X = cms.int32( 1 ),
+    ROWS_PER_ROC = cms.int32( 80 ),
+    COLS_PER_ROC = cms.int32( 52 )
+  ),
+  applyAlignment = cms.bool( True ),
+  alignmentsLabel = cms.string( "" )
+)
```
Comment (ianna) [inline Geometry/TrackerGeometryBuilder/python/trackerDigiGeometryESModule_cfi.py]: This file is not missing as it is auto-generated. See 
http://cmslxr.fnal.gov/source/Geometry/TrackerGeometryBuilder/plugins/TrackerDigiGeometryESModule.cc?v=CMSSW_7_4_X_2015-01-15-0200#0049
This newly introduced file could be a reason for RelVal failures.

**Comment by ianna** [comment]
-1

File: Geometry/TrackerGeometryBuilder/python/trackerDigiGeometryESModule_cfi.py (hunk 2 above)
Comment (mtosi) [inline Geometry/TrackerGeometryBuilder/python/trackerDigiGeometryESModule_cfi.py]: yes, but @gennai asked for having a physical file

Comment (fwyzard) [inline Geometry/TrackerGeometryBuilder/python/trackerDigiGeometryESModule_cfi.py]: Thou shalt not introduce cfi files, if they are automatically produced by
fillDescriptions.

.Andrea

Comment (Dr15Jones) [inline Geometry/TrackerGeometryBuilder/python/trackerDigiGeometryESModule_cfi.py]: agreed.

**Comment by mtosi** [comment]
ciao,
thanks for pointing out this issue

I deleted few python config files
but I left 
TrackingTools/Producers/python/analyticalPropagatorESProducer_cff.py
TrackingTools/Producers/python/trajectoryCleanerESProducer_cfi.py

please, let me know if they are fine
thanks !

**Comment by ianna** [comment]
@mtosi - thanks! The PR does not require Geometry signature now.

**Comment by slava77** [comment]
@cmsbuild please test

File: TrackingTools/Producers/python/trajectoryCleanerESProducer_cfi.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import FWCore.ParameterSet.Config as cms
+
+## TEMPLATE as it is used by HLT
+trajectoryCleanerBySharedHits = cms.ESProducer( "TrajectoryCleanerESProducer",
```
Comment (slava77) [inline TrackingTools/Producers/python/trajectoryCleanerESProducer_cfi.py]: what are we supposed to do with 
https://github.com/cms-sw/cmssw/blob/CMSSW_7_4_0_pre5/TrackingTools/TrajectoryCleaning/python/TrajectoryCleanerBySharedHits_cfi.py
``` trajectoryCleanerBySharedHits = cms.ESProducer("TrajectoryCleanerESProducer", ```
?
the same name, but in a different place and with different parameters.

Note that the style rules for cfi files is to define modules matching the name of the file (trajectoryCleanerESProducer_cfi vs trajectoryCleanerBySharedHits)

**Comment by slava77** [comment]
uhm, older PRs need manual open of the PR page to see if there was an update.
Thanks for dropping the file

**Comment by slava77** [comment]
+1

for 
https://github.com/cms-sw/cmssw/pull/71939    b64ff9
only one new file is added and it's the same as in the last commit already checked by jenkins

**Pull request merged**